packages/apollo/src/updaters/removeFromList.ts

```diff
@@ -1,22 +1,20 @@
 import { DocumentNode } from 'graphql';
-import get from 'lodash/get';
-import set from 'lodash/set';
-import { DataProxy } from 'apollo-cache';
+import { get, set } from 'lodash/fp';
+import { DataProxy, ApolloCache, StoreObject } from '@apollo/client';
 import prepareQuery from '../utils/prepareQuery';
 import getQueryName from '../utils/getQueryName';
 
 export default function removeFromList<Result, Vars = {}>(
-  docOrQuery: DocumentNode | DataProxy.Query<Vars>,
+  docOrQuery: DocumentNode | DataProxy.Query<Vars, Result>,
   listPath: string,
   id: string | number,
   idName: string = 'id',
 ) {
-  const query = prepareQuery<Vars>(docOrQuery);
+  const query = prepareQuery<Result, Vars>(docOrQuery);
 
-  return (cache: DataProxy) => {
+  return (cache: ApolloCache<Result>) => {
```

Comment:
`ApolloCache` class implements `DataProxy` interface and we need it to evict cache